helix-core/src/main/java/org/apache/helix/manager/zk/ZkClient.java

```diff
@@ -55,33 +49,26 @@
  * TODO Completely replace usage of the raw ZkClient within helix-core. Instead, using HelixZkClient. --JJ
  */
 
-public class ZkClient extends org.apache.helix.manager.zk.zookeeper.ZkClient implements HelixZkClient {
-  private static Logger LOG = LoggerFactory.getLogger(ZkClient.class);
-
-  public static final int DEFAULT_OPERATION_TIMEOUT = Integer.MAX_VALUE;
-  public static final int DEFAULT_CONNECTION_TIMEOUT = 60 * 1000;
-  public static final int DEFAULT_SESSION_TIMEOUT = 30 * 1000;
-
+@Deprecated
+public class ZkClient extends org.apache.helix.zookeeper.impl.client.ZkClient {
   /**
-   *
-   * @param zkConnection
+   *  @param zkConnection
    *            The Zookeeper connection
    * @param connectionTimeout
    *            The connection timeout in milli seconds
-   * @param zkSerializer
-   *            The Zookeeper data serializer
    * @param operationRetryTimeout
    *            Most operations are retried in cases like connection loss with the Zookeeper servers. During such failures, this
    *            <code>operationRetryTimeout</code> decides the maximum amount of time, in milli seconds, each
    *            operation is retried. A value lesser than 0 is considered as
    *            "retry forever until a connection has been reestablished".
+   * @param zkSerializer
+   *            The Zookeeper data serializer
    * @param monitorType
    * @param monitorKey
    * @param monitorInstanceName
-   *            These 3 inputs are used to name JMX monitor bean name for this ZkClient.
+   *            These 3 inputs are used to name JMX monitor bean name for this RealmAwareZkClient.
    *            The JMX bean name will be: HelixZkClient.monitorType.monitorKey.monitorInstanceName.
    * @param monitorRootPathOnly
-   *            Should only stat of access to root path be reported to JMX bean or path-specific stat be reported too.
    */
   public ZkClient(IZkConnection zkConnection, int connectionTimeout, long operationRetryTimeout,
```

Comment:
Do we still need these constructors in old file? Or we can just leave the class purely extend what we have in zookeeper-api module?

Reply:
Don't you think we should keep all public constructors for backward-compatibility?

Reply:
Oh. The move also involve some constructor change? Then I am OK for that.